document/core/exec/instructions.rst

```diff
@@ -182,6 +182,489 @@ Where the underlying operators are non-deterministic, because they may return on
+     and finally packing the resulting sequence :math:`\simdof_{\I32\K{x}4}(m_1~m_2~m_3~m_4)`.
+
+
+.. _exec-simd-const:
+
+:math:`\V128\K{.}\CONST~c`
+..........................
+
+1. Push the value :math:`\V128.\CONST~c` to the stack.
+
+.. note::
+   No formal reduction rule is required for this instruction, since |CONST| instructions coincide with :ref:`values <syntax-val>`.
+
+
+.. _exec-simd-vsunop:
+
+:math:`\V128\K{.}\vsunop`
+.........................
+
+1. Assert: due to :ref:`validation <valid-vsunop>`, a value of :ref:`value type <syntax-valtype>` |V128| is on the top of the stack.
+
+2. Pop the value :math:`\V128.\CONST~c_1` from the stack.
+
+3. Let :math:`c` be the result of computing :math:`\vsunop_{\I128}(c_1)`.
+
+4. Push the value :math:`\V128.\CONST~c` to the stack.
+
+.. math::
+   \begin{array}{lcl@{\qquad}l}
+   (\V128\K{.}\CONST~c_1)~\V128\K{.}\vsunop &\stepto& (\V128\K{.}\CONST~c)
+     & (\iff c = \vsunop_{\I128}(c_1)) \\
+   \end{array}
+
+
+.. _exec-simd-vsbinop:
+
+:math:`\V128\K{.}\vsbinop`
+..........................
+
+1. Assert: due to :ref:`validation <valid-vsbinop>`, two values of :ref:`value type <syntax-valtype>` |V128| are on the top of the stack.
+
+2. Pop the value :math:`\V128.\CONST~c_2` from the stack.
+
+3. Pop the value :math:`\V128.\CONST~c_1` from the stack.
+
+4. Let :math:`c` be the result of computing :math:`\vsbinop_{\I128}(c_1, c_2)`.
+
+5. Push the value :math:`\V128.\CONST~c` to the stack.
+
+.. math::
+   \begin{array}{lcl@{\qquad}l}
+   (\V128\K{.}\CONST~c_1)~(\V128\K{.}\CONST~c_2)~\V128\K{.}\vsbinop &\stepto& (\V128\K{.}\CONST~c)
+     & (\iff c = \vsbinop_{\I128}(c_1, c_2)) \\
+   \end{array}
+
+
+.. _exec-simd-vsternop:
+
+:math:`\V128\K{.}\vsternop`
+...........................
+
+1. Assert: due to :ref:`validation <valid-vsternop>`, three values of :ref:`value type <syntax-valtype>` |V128| are on the top of the stack.
+
+2. Pop the value :math:`\V128.\CONST~c_3` from the stack.
+
+3. Pop the value :math:`\V128.\CONST~c_2` from the stack.
+
+4. Pop the value :math:`\V128.\CONST~c_1` from the stack.
+
+5. Let :math:`c` be the result of computing :math:`\vsternop_{\I128}(c_1, c_2, c_3)`.
+
+6. Push the value :math:`\V128.\CONST~c` to the stack.
+
+.. math::
+   \begin{array}{lcl@{\qquad}l}
+   (\V128\K{.}\CONST~c_1)~(\V128\K{.}\CONST~c_2)~(\V128\K{.}\CONST~c_3)~\V128\K{.}\vsternop &\stepto& (\V128\K{.}\CONST~c)
+     & (\iff c = \vsternop_{\I128}(c_1, c_2, c_3)) \\
+   \end{array}
+
+
+.. todo::
+   shuffle, load splat, load extend.
+
+
+.. _exec-simd-splat:
+
+:math:`\shape\K{.}\SPLAT`
+.........................
+
+1. Let :math:`t` be the type :math:`\unpacked(\shape)`.
+
+2. Assert: due to :ref:`validation <valid-simd-splat>`, a value of :ref:`value type <syntax-valtype>` :math:`t` is on the top of the stack.
+
+3. Pop the value :math:`t.\CONST~c_1` from the stack.
+
+4. Let :math:`n` be the integer :math:`\lanes(\shape)`.
```

Comment:
below `N` is often used for the lane count, maybe use that instead for consistency?